src/main/java/pl/javastart/task/Main.java

```diff
@@ -1,8 +1,172 @@
 package pl.javastart.task;
 
+import java.time.*;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+import java.time.temporal.TemporalUnit;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+
 public class Main {
+    private static final List<DateTimeFormatter> DATE_TIME_PATTERNS = List.of(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"),
+            DateTimeFormatter.ofPattern("d.MM.yyyy HH:mm:ss"));
+    private static final List<DateTimeFormatter> DATE_PATTERNS = List.of(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+    private static final DateTimeFormatter DISPLAYED_DATE_PATTERN = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+    private static final char PLUS = '+';
+    private static final char MINUS = '-';
+    private Map<String, ZoneId> zones;
+
+    public Main() {
+        this.zones = createZonesMap();
+    }
 
     public static void main(String[] args) {
+        Main main = new Main();
+        main.run(new Scanner(System.in));
+    }
+
+    public void run(Scanner scanner) {
+        LocalDateTime dateTime = loadTimeInput(scanner);
+        if (dateTime == null) {
+            System.out.println("Niepoprawna data");
+        } else {
+            showTime(dateTime);
+        }
+    }
+
+    private LocalDateTime loadTimeInput(Scanner scanner) {
+        System.out.println("Podaj datę/różnicę czasu:");
+        String input = scanner.nextLine();
+        if (input.startsWith("t")) {
+            return calculateDateFromNow(input.substring(1));
+        } else {
+            return adjustDate(input);
+        }
+    }
+
+    private LocalDateTime calculateDateFromNow(String timeShift) {
+        LocalDateTime now = LocalDateTime.now();
+        int indexOfPlus = timeShift.indexOf(PLUS);
+        int indexOfMinus = timeShift.indexOf(MINUS);
+        while (indexOfPlus != -1 || indexOfMinus != -1) {
+            int number = 0;
+            char operation = PLUS;
+            int timeUnitIndex = 0;
+            if (indexOfMinus == -1 || indexOfPlus != -1 && indexOfPlus < indexOfMinus) {
+                number = getTimeNumberFromString(timeShift.substring(indexOfPlus + 1));
+                operation = PLUS;
+                timeUnitIndex = indexOfPlus + Integer.toString(number).length() + 1;
+            } else if (indexOfPlus == -1 || indexOfMinus != -1 && indexOfMinus < indexOfPlus) {
+                number = getTimeNumberFromString(timeShift.substring(indexOfMinus + 1));
+                operation = MINUS;
+                timeUnitIndex = indexOfMinus + Integer.toString(number).length() + 1;
+            }
```

Comment:
W tym if-else robisz praktycznie to samo, jedynie zmienia się znak PLUS/MINUS
Można to uprościć